### DISCUSSION TITLE
Attribute Error on Mobile (android) scroll

### DISCUSSION THREAD

**Discussion by chad63e** [anvilistas/anvil-extras#324] [Q&A]
I am getting this error when scrolling on my mobile phone:

```
AttributeError: 'HTMLDocument' object has no attribute 'closest' at app/anvil_extras/popover.py, line 399
```

**Comment by s-cork** [comment]
What version is your OS on your mobile?
And which browser?

**Comment by chad63e** [comment]
Android v 12

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by s-cork** [reply]
And are you able to check what version of chrome you're using on the phone?

**Reply by s-cork** [reply]
Or if it's a different browser. The version of that browser.

**Comment by chad63e** [comment]
I was using Brave. See attachment

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by s-cork** [reply]
Sending attachments doesn't work from email to GitHub discussions. 


Worth downloading the GitHub app or opening the discussion in a browser.

**Comment by chad63e** [comment]
Brave v 1.37.116

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by s-cork** [reply]
I think I've addressed this in PR #325 
Whenever the UK wakes up then it should go live ready for you to test

**Comment by chad63e** [comment]
Thanks Stu,

I'll check it out in the morning. Have a good night.

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by chad63e** [comment]
Not sure if this update has gone live yet or not, but I am still experiencing the error.  Also, It is happening on my windows pc using Chrome v 100.0.4896.127 (64 bit).

I am not sure if this might have anything to do with it, but the element that I am attaching the popover is a "circle" element within an "svg".  

I have even tried adding an event listener on the parent page to trigger the popover to manually "hide" on scroll.  This manual trigger does cause the popover to dismiss, but the error persists.

**Reply by s-cork** [reply]
what's the current error including line number?

**Reply by chad63e** [reply]
```
AttributeError: 'HTMLDocument' object has no attribute 'closest' at app/anvil_extras/popover.py:399
```

**Reply by s-cork** [reply]
wanna try again?

**Reply by chad63e** [reply]
I am still getting the same error?

**Reply by chad63e** [reply]
It's not doing it in the "Development" version

**Reply by s-cork** [reply]
ok - just double check you're on the development branch and refresh the cache
I'll need a clone please
or a link to the live app

**Reply by s-cork** [reply]
Changes will only be on the Development version

**Reply by s-cork** [reply]
Did you check to development version earlier? or only just now

**Reply by chad63e** [reply]
Now, It is giving me some odd behavior where if I have scrolled and it "toggles" the popover, the next time I click on the icon to get a new popover, it is not in the proper placement?  See attached
<img width="399" alt="popover scroll issue" src="https://user-images.githubusercontent.com/46352497/164753597-7628d29d-ea90-4a08-8d3b-aaf58dc35fd2.PNG">
.

I've published the app at the present development for this issue.

**Reply by s-cork** [reply]
sorry - I don't really have enough information to help with that one.
There's an update method you can call
`pop(dom_node, "update")`

**Comment by chad63e** [comment]
Ok, i'll give a try

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by chad63e** [reply]
After some further digging, it appears that "update" and has_popover do not work with dom nodes.  I was able to resolve the incorrect popover location by linking it to an Anvil component and not a dom node.  Also, has_popover throws an error.  I was able to reproduce the has_popover functionality with this code:

```
popover_id = my_dom_node.getAttribute("popover_id") or ""
has_popover = True if len(popover_id) > 0 else False
```

**Reply by s-cork** [reply]
> Also, has_popover throws an error.

More context please. Is this only with a dom node? Or always or sometimes or when?

**Comment by chad63e** [comment]
I believe only with dom nodes

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by s-cork** [reply]
I think given that the code now works and although I said it should work accidentally with domnodes. 

I guess it doesn't. 
So you're current approach is the right one.

Unless you're hindered by the current approach I don't indent to officially support domnodes.

**Comment by chad63e** [comment]
I think my current work around will work for now.

Hopefully down the road this will be supported or some way to make a dom node an anvil component.

Thanks for all the help!

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by lewysigns** [comment]
This error is popping up for me on chrome in the IDE and Production. It only pops up when scrolling, if the pop over is shown.

The application:

```
 self.content_panel.popover(content="User needs to be assigned", trigger="hover", placement="top")
```

**Reply by s-cork** [reply]
@lewysigns - Are you on development or published?

Please check the dev dependency. 
If it's still broken could you provide a clone link and reproduction steps.

**Reply by lewysigns** [reply]
I was on published.

development does fix it. Thank you!

**Reply by lewysigns** [reply]
Should Have searched #324 in the issues! sorry, still learning Github infrastructure.
